### PR TITLE
Tests/setup tests

### PR DESCRIPTION
**Pull request by bishwarup307**
Implements tests, coverage and codecov.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+"""
+__author__: HashTagML
+license: MIT
+Created: Thursday, 1st January 1970
```
Comment (aman0044) [inline tests/conftest.py:4]: Are you a time traveller?

**Review by bharatkumarreddy — APPROVED** [review]
Looks good

**Review by aman0044 — APPROVED** [review]
Looks to good to me.

File: tests/conftest.py (hunk 1 above)
Comment (bishwarup307) [reply tests/conftest.py:4]: Lol, no idea what happened there.

File: optical/converter/utils.py [hunk 2]
```diff
@@ -244,8 +244,11 @@ def get_id_to_class_map(df: pd.DataFrame):
 def find_splits(image_dir: Union[str, os.PathLike], annotation_dir: Union[str, os.PathLike], format: str):
     """find the splits in the dataset, will ignore splits for which no annotation is found"""
 
+    # print(f"passed format: {format}")
+
     exts = {
         "coco": "json",
+        "csv": "csv",
```
Comment (satheeshkatipomu) [inline optical/converter/utils.py:251]: duplicate `csv` record

**Comment by satheeshkatipomu** [comment]
@bishwarup307 , need an option to run tests on local through `tox`?

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+from typing import Union
+import pytest
+import numpy as np
+import pandas as pd
+
+import os
+import shutil
+
+from kaggle.api.kaggle_api_extended import KaggleApi
+
```
Comment (satheeshkatipomu) [inline tests/conftest.py:17]: I think we need to check `tensorflow` installation, before expecting it while running test case

File: tests/test_optical.py [hunk 4]
```diff
@@ -1,5 +1,111 @@
...
+    convert_pascal,
+    convert_createml,
+    convert_sagemaker,
+    convert_tfrecord,
+    convert_simple_json,
+)
+
+import pandas as pd
+import altair as alt
+
```
Comment (satheeshkatipomu) [inline tests/test_optical.py:22]: I think we need to check tensorflow installation, before expecting it while running test case

File: tests/test_optical.py [hunk 5]
```diff
@@ -1,5 +1,111 @@
...
+    assert set(splits) == set(dirs[2])
+    assert has_im_split == dirs[3]
+
+
+@pytest.mark.parametrize("fmt", ["coco", "createml", "sagemaker", "yolo", "pascal", "csv", "tfrecord", "simple_json"])
+@pytest.mark.parametrize("has_split", [False, True])
+def test_resolve_df(root, fmt, has_split):
+
+    annotation = Annotation(root, format=fmt)
```
Comment (satheeshkatipomu) [inline tests/test_optical.py:50]: Test case failing when `tensorflow` is not installed. suggest below change.
```
if fmt == "tfrecord" and not _TF_INSTALLED:
        error = "Please Install Tensorflow for tfrecord support"
        with pytest.raises(ImportError, match=error):
            _ = Annotation(root, format=fmt)
        return
```

File: tests/test_optical.py [hunk 6]
```diff
@@ -1,5 +1,111 @@
...
+    assert isinstance(scatter, alt.Chart)
+
+
+@pytest.mark.parametrize("fmt", ["coco"])
+@pytest.mark.parametrize("has_split", [True])
+def test_tfrecord_export(root, fmt, test_dir):
+    annotation = Annotation(root, format=fmt)
+
+    output_dir = Path(test_dir).joinpath("exports")
+    annotation.export(to="tfrecord", output_dir=output_dir)
```
Comment (satheeshkatipomu) [inline tests/test_optical.py:108]: Need to check `tensorfow` installation as mentioned above.

**Comment by bishwarup307** [comment]
@satheeshkatipomu I think all the extras should be necessarily tested and it shouldn't be made optional. Even though they are extras we need to make sure they work and as such skipping those tests is not a good choice. We can always invoke tensorflow (and any other extras) installation in the test env and hence run all the tests.

**Review by satheeshkatipomu — APPROVED** [review]
Looks good.

**Comment by bishwarup307** [comment]
there were some problem with the cached poetry env in the macos runner. fixed now. merging this one.

**Pull request merged**